internal/services/mssql/mssql_server_resource.go

```diff
@@ -443,18 +388,80 @@ func resourceMsSqlServerUpdate(d *pluginsdk.ResourceData, meta interface{}) erro
 				return fmt.Errorf("deleting Azure Active Directory Administrator %s: %+v", id, err)
 			}
 		}
+
+		if aadOnlyAuthentictionsEnabled {
+			aadOnlyAuthentictionsProps := serverazureadonlyauthentications.ServerAzureADOnlyAuthentication{
+				Properties: &serverazureadonlyauthentications.AzureADOnlyAuthProperties{
+					AzureADOnlyAuthentication: aadOnlyAuthentictionsEnabled,
+				},
+			}
+
+			err := aadOnlyAuthenticationsClient.CreateOrUpdateThenPoll(ctx, *id, aadOnlyAuthentictionsProps)
+			if err != nil {
+				return fmt.Errorf("updating Azure Active Directory Only Authentication for %s: %+v", id, err)
+			}
+		}
 	}
 
-	if aadOnlyAuthentictionsEnabled := expandMsSqlServerAADOnlyAuthentictions(d.Get("azuread_administrator").([]interface{})); d.HasChange("azuread_administrator") && aadOnlyAuthentictionsEnabled {
-		aadOnlyAuthentictionsProps := serverazureadonlyauthentications.ServerAzureADOnlyAuthentication{
-			Properties: &serverazureadonlyauthentications.AzureADOnlyAuthProperties{
-				AzureADOnlyAuthentication: aadOnlyAuthentictionsEnabled,
-			},
+	if payload := existing.Model; payload != nil {
```

Comment:
Sorry this may be a silly question but was there a reason for moving this part of the code?

Reply:
Moving this part of the code is key to solving issue #25116, where the Azure Rest API does not allow `administrator_login_password` updates when `azuread_authentication_only = true`. Because the API has such limitations, the correct behavior of TF should be to first update `azuread_authentication_only = false` (specified by the user), and then update `administrator_login_password`, so that the update can be successful.